crates/runtime/src/accelerated_table/refresh.rs

```diff
@@ -234,6 +234,31 @@ impl Refresh {
 
         Ok(())
     }
+
+    pub(crate) async fn should_refresh(
+        &self,
+        last_checkpoint: Option<Arc<dyn DatasetCheckpointer>>,
+    ) -> bool {
+        // If there is no checkpoint, we need to start a refresh.
+        let Some(last_checkpoint) = last_checkpoint else {
+            return true;
+        };
+
+        let has_previous_checkpoint = match self.mode {
+            RefreshMode::Full => last_checkpoint.exists().await,
+            // Append and Changes modes are always refreshed since they stream changes from the source table.
+            RefreshMode::Append | RefreshMode::Changes => return true,
+            RefreshMode::Disabled => return false,
+        };
+
+        // If there is no previous checkpoint, we need to start a refresh.
+        if !has_previous_checkpoint {
+            return true;
+        }
+
+        // If we have a previous checkpoint and the refresh interval is set, then we need to refresh - otherwise we don't.
```

Comment:
This is a misleading comment. We should note this is temporary behavior, in the future, we need to check the interval time against the current time to determine if we need to refresh.